llvm/include/llvm/Analysis/TargetTransformInfoImpl.h

```diff
@@ -781,7 +781,9 @@ class TargetTransformInfoImplBase {
     default:
       break;
     case Intrinsic::experimental_vector_histogram_add:
-      // For now, we want explicit support from the target for histograms.
+    case Intrinsic::experimental_vector_match:
+      // For now, we want explicit support from the target for histograms and
+      // matches.
```

Comment:
This seems excessive given you have implemented common lowering code where I think the base cost is effectively:

`(cost_of_extract_elt+cost_of_vector_splat+cost_of_vector_cmp+)*number_of_needle_elements`.

Reply:
Thanks very much for the feedback, I've now implemented this (sorry for the delay, last week was a bit busy). Could you please have a look and let me know if what I did is what you had in mind?